sdk/python/kfp/v2/compiler/compiler_utils.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""KFP v2 DSL compiler utility functions."""
+
+from typing import List, Mapping
+from kfp.v2 import dsl
+from kfp.v2.proto import pipeline_spec_pb2
+
+
+def build_runtime_parameter_spec(
+    pipeline_params: List[dsl.PipelineParam]
+) -> Mapping[str, pipeline_spec_pb2.PipelineSpec.RuntimeParameter]:
```

Comment:
nit/question to learn: why did you go for mapping her for the typing instead of Dict?
```suggestion
) -> Mapping[str, pipeline_spec_pb2.PipelineSpec.RuntimeParameter]:
```

Reply:
+1 for `Dict`

More restrictive type hint will be better for return types, while it's generally the other way around for parameter types

Reply:
`Dict` or `Mapping` actually doesn't matter too much here. I've seen arguments in both directions, and abstract type may still be favored even for return type -- it gives flexibility to change implementation without breaking downstream. 
The main breaking point for me to choose `Mapping` over `Dict` here is the implication of immutable vs. mutable. The returned spec should be considered final and immutable.

Reply:
> The returned spec should be considered final and immutable.

Makes sense. Thanks for the explanation!